src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -213,19 +250,34 @@ private ImmutableArray<string> DecodeSuppressMessageAttributes(ISymbol symbol)
                     continue;
                 }
 
-                builder.Add(info.Id);
+                AddOrUpdate(info, builder);
             }
 
-            return builder.ToImmutableAndFree();
+            return builder.ToImmutable();
         }
 
-        private static void DecodeGlobalSuppressMessageAttributes(Compilation compilation, ISymbol symbol, ISymbol suppressMessageAttribute, GlobalSuppressions globalSuppressions)
+        private static void AddOrUpdate(SuppressMessageInfo info, IDictionary<string, SuppressMessageInfo> builder)
+        {
+            // TODO: How should we deal with multiple SuppressMessage attributes, with different suppression info/states?
```

Comment:
Well this comment was valid when we had workflow states and multiple SuppressMessageAttributes with different workflow state could conflict. I'll remove this redundant comment.